README.md

```diff
@@ -296,6 +296,27 @@ Or:
     ]
 ```
 
+* `customActions`: (optional) Defines custom actions to be executed at various events during the generated sync function's execution. Specified as an object where each property specifies a JavaScript function to be executed when the corresponding event is completed. In each case, the function accepts as parameters (1) the new document and (2) the old document that is being replaced/deleted (if any). In cases where the document is in the process of being deleted, the first parameter's `_deleted` property will be `true`, so be sure to account for such cases. If the document does not yet exist, the second parameter will be null or undefined and, in some cases where the document previously existed (i.e. it was deleted), the second parameter _may_ be non-null and its `_deleted` property will be `true`. Custom actions may call functions from the [standard sync function API](http://developer.couchbase.com/documentation/mobile/current/guides/sync-gateway/sync-function-api-guide/index.html) (e.g. `requireAccess`, `requireRole`, `requireUser`, `access`, `role`, `channel`) and may indicate errors via the `throw` function to prevent the document from being written. The custom action events that are available:
```

Comment:
I elected to pass the `oldDoc` parameter completely unmodified instead of replacing it with `null` if its `_deleted` property is `true` to allow for the greatest level of flexibility for custom actions.